cloud_guardian/iam_model/graph/main.py

```diff
@@ -75,18 +98,84 @@
     group = GroupFactory.from_dict(group_data)
     for user_data in group_data["Users"]:
         user = UserFactory._instances[user_data["UserArn"]]
-        is_part_of_relationship = IsPartOf(user, group)
-        graph.add_relationship(is_part_of_relationship)
+        graph.add_relationship(IsPartOf(user, group))
 
 
-# TODO: Parse the rest of the relationships (see relationships.py)
+# Can Assume Role relationships
+for role_data in data["roles.json"]["Roles"]:
+    role = RoleFactory.from_dict(role_data)
+    for statement in role_data["AssumeRolePolicyDocument"]["Statement"]:
+        if statement["Effect"] != "Allow":
+            continue
+
+        # TODO: implement a mechanism to ensure that the list of users is complete
+        try:
+            user = UserFactory._instances[statement["Principal"]["AWS"]]
+        except KeyError as e:
+            logger.error(f"user does not exist {e}")
+
+        graph.add_relationship(CanAssumeRole(user, role))
+        logger.info(
+            f"added relationship between {user.user_name} and role {role.role_name}: can assume role"
+        )
 
-# roles.json contain all you need to make the "CanAssumeRole" relationships
+# Users permissions
+for user_data in data["users.json"]["Users"]:
+    # TODO: implement a mechanism to ensure that the list of users is complete
+    try:
+        user = UserFactory._instances[user_data["Arn"]]
+    except KeyError as e:
+        logger.error(f"user does not exist {e}")
+
+    for policy_data in user_data["AttachedPolicies"]:
+        arn = policy_data["PolicyArn"]
+        policy = extract_policy_from_ARN(arn)
+
+        permission = PermissionFactory.get_or_create(
+            action=ActionFactory.get_or_create(arn),
+            effect=Effect(
+                Effect.ALLOW
+            ),  # NOTE: do we assume all permissions ALLOW by default?
+            conditions=[],
+        )
+
+        # FIXME: how do we define the target here? (I am temporarily using `None`)
+        graph.add_relationship(HasPermission(user, target=None, permission=permission))
+        logger.info(
+            f"added relationship between {user.user_name} and {None}: {permission.action} [{permission.effect}]"
+        )
+
+# Groups
+for group_data in data["groups.json"]["Groups"]:
+    permissions: List[Permission] = []
+    users_belonging_to_group: List[User] = []
+
+    for group_policy in group_data["AttachedPolicies"]:
+        permission = PermissionFactory.get_or_create(
```

Comment:
you can use directly PermissionFactory.from_dict and pass the Statement dict

Reply:
If I am not mistaken that would not work as `groups.json` does not specify the effects.

Reply:
groups.json and users.json just tells you the ID of the policy, then you match the ID in the identites_policies and you parse the "Statement" with PermissionFactory, no?